### PR TITLE
geckodriver: 0.26.0 -> 0.29.1

### PR DESCRIPTION
**Pull request by SamirTalwar**
###### Motivation for this change

Upgrading geckodriver to the latest version.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119164` at 2d7952cd run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/64a383d39988626ddc136eadd0ef511d">geckodriver</a></li>
    <li><a href="https://gist.github.com/b2463274b920d99c0a3205fa47da1a49">python38Packages.robotframework-selenium2library</a></li>
    <li><a href="https://gist.github.com/925655bb4a87d5260dbfb8362e1ba1be">python38Packages.robotframework-seleniumlibrary</a></li>
    <li><a href="https://gist.github.com/540a24db62a11e64ca573fc6e89f7d1d">python38Packages.selenium</a></li>
    <li><a href="https://gist.github.com/4229989a03a5ef0e193cd394ae943048">python38Packages.splinter</a></li>
    <li><a href="https://gist.github.com/a7627477eccf47502307f2ef932f93b5">python39Packages.robotframework-selenium2library</a></li>
    <li><a href="https://gist.github.com/a62c0e55e090f7b409b764e9b0d753dc">python39Packages.robotframework-seleniumlibrary</a></li>
    <li><a href="https://gist.github.com/53fb7a557e78502773575fd80ab3d6c1">python39Packages.selenium</a></li>
    <li><a href="https://gist.github.com/267851fd0899e20c81faa969379982dd">python39Packages.splinter</a></li>
    <li><a href="https://gist.github.com/471c104c1fbb6e68154eb97a80cf1978">searx</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by SamirTalwar** [comment]
I'm really not sure why this works locally for me but gets a hash mismatch on the Cargo build in CI.

```
hash mismatch in fixed-output derivation '/nix/store/gjzq1djpxxvl0l737cshglpkkl3af48i-geckodriver-0.29.1-vendor.tar.gz':
  wanted: sha256:1fxm4jgg95igqadi97dzhrw7bhi5hfh7ww81c3z4yk0lznbzv6mx
  got:    sha256:1vajlcpyk77v6nvhs737yi8hs7ids9kz0sbwy29rm1vmmfjp2b27
```

**Comment by stephank** [comment]
I also see that hash mismatch, with the same output as CI. Maybe try regenerate it locally?

```sh
nix-store --delete /nix/store/gjzq1djpxxvl0l737cshglpkkl3af48i-geckodriver-0.29.1-vendor.tar.gz
nix-build -A geckodriver
```

File: pkgs/development/tools/geckodriver/default.nix [hunk 1]
```diff
@@ -3,27 +3,28 @@
...
     # normally guessed by the url's file extension, force it to unpack properly
     unpackCmd = "unzip $curSrc";
   });
 
   cargoPatches = [ ./cargo-lock.patch ];
-  cargoSha256 = "1dv8vcjy8r9z19grj4gms05bhaafyr42y3q69h3azwq6dmacfd3y";
+  cargoSha256 = "1fxm4jgg95igqadi97dzhrw7bhi5hfh7ww81c3z4yk0lznbzv6mx";
 
-  buildInputs = lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
+  buildInputs = [ libiconv ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
```
Comment (stephank) [inline pkgs/development/tools/geckodriver/default.nix:27]:
```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
```

libiconv is often a Darwin only dependency. This builds fine without it on Linux.

**Review by stephank — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/geckodriver/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/geckodriver/default.nix:27]:
```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [ libiconv Security ];
```
darwin shouldn't be an input as it cannot be easily overwriten.

Comment (SamirTalwar) [reply pkgs/development/tools/geckodriver/default.nix:27]: @SuperSandro2000: I can't make `Security` an input here, at least not just by updating the parameters at the top. Any idea how I can make it injectable directly?

Comment (stephank) [reply pkgs/development/tools/geckodriver/default.nix]: See for example: https://github.com/NixOS/nixpkgs/blob/0c896e754d968655826d4eeb319dde68cfb33b39/pkgs/top-level/all-packages.nix#L746-L748

Comment (SamirTalwar) [reply pkgs/development/tools/geckodriver/default.nix]: That works. Pushed, thanks!

**Comment by SamirTalwar** [comment]
It's looking pretty healthy now, I think.

**Review by stephank — APPROVED** [review]

**Review by jraygauthier — APPROVED** [review]
LGTM code-wise. Did not attempt local build nor use.

**Comment by SamirTalwar** [comment]
Thanks for merging, and thanks for teaching me a bit more about Nix. 😁

**Pull request merged**